source/code/include/playfab/PlayFabPlatformUtils.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <sstream>
+
+namespace PlayFab
+{
+    typedef std::chrono::time_point<std::chrono::system_clock> TimePoint;
+
+    inline TimePoint GetTimePointNow()
+    {
+        return std::chrono::system_clock::now();
+    }
+
+    inline time_t GetPlayFabTimeTNow()
+    {
+        time_t now = std::chrono::system_clock::to_time_t(GetTimePointNow());
+#if defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_LINUX) || defined(PLAYFAB_PLATFORM_SWITCH)
+        return static_cast<Json::Int64>(now);
+#else // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_SWITCH
+        return now;
+#endif // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_SWITCH
+    }
+
+    inline void AppendIntToString(int value, std::string& output)
+    {
+        // itoa is not available in android
+        char buffer[16];
+#if defined(PLAYFAB_PLATFORM_IOS) || defined(PLAYFAB_PLATFORM_ANDROID) || defined(PLAYFAB_PLATFORM_LINUX) || defined(PLAYFAB_PLATFORM_PLAYSTATION) || defined(PLAYFAB_PLATFORM_SWITCH)
+        sprintf(buffer, "%d", value);
+#else // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_PLAYSTATION || PLAYFAB_PLATFORM_SWITCH
+        sprintf_s(buffer, "%d", value);
+#endif // PLAYFAB_PLATFORM_IOS || PLAYFAB_PLATFORM_ANDROID || PLAYFAB_PLATFORM_LINUX || PLAYFAB_PLATFORM_PLAYSTATION || PLAYFAB_PLATFORM_SWITCH
+        output.append(buffer);
+    }
+
+    inline std::string LocalTimeTToUtcString(time_t now)
```

Comment:
>LocalTime [](start = 23, length = 9)

In the previous iteration of this change we discussed how I don't think "Local" is the right term to use here.

You said in your experiments that it always represented a local time.

I'm curious what your experiments were... you said "In my past experiments, parsing and printing the output of time_t required shifting/correcting for the timezone, given the input string."

what were you printing? some stringified form of the time_t? the time_t itself as an int or something?

Reply:
https://github.com/PlayFab/XPlatCoreTemplate/blob/pg/timestamp/source/test/TestApp/PlayFabApiTest.cpp#L287

I can't say exactly what I did/tested ages ago.
The test above is the current test, and satisfying that test should be sufficient.

That said, I'm hours deep into testing this change, so I'm very hesitant to make changes to this PR. I'm happy to file a task for a non-breaking redesign/rename/fix.

Reply:
While this PR gets all the projects to compile and tests to pass for the new PlatformUtils.h file, I agree the PlatformUtils.h file itself isn't done.
bug filed.